File: _Src/Freem/Freem.Web.Api.Public.Client/Implementations/StatisticsClient.cs
using System.Text.Json;
using Freem.Http.Requests.Abstractions;
using Freem.Http.Requests.Entities;
using Freem.Web.Api.Public.Client.Implementations.Base;
using Freem.Web.Api.Public.Client.Models;
using Freem.Web.Api.Public.Contracts.DTO.Statistics;

namespace Freem.Web.Api.Public.Client.Implementations;

public sealed class StatisticsClient : BaseClient
{
    public StatisticsClient(IHttpClient client, JsonSerializerOptions options, TokenLoader tokenLoader) 
        : base(client, options, tokenLoader)
    {
    }

    public Task<ClientResult<StatisticsPerDaysResponse>> GetAsync(
        StatisticsPerDaysRequest query, CancellationToken cancellationToken = default)
    {
        var request = HttpRequest.Get("api/v1/statistics/per-days")
            .WithQueryParameter(nameof(query.Period), query.Period.ToString());

        return SendAuthorizedAsync<StatisticsPerDaysResponse>(request, cancellationToken);
    }
}

File: _Src/Freem/Freem.Web/src/clients/StatisticsClient.ts
import {IHttpClient} from "./http/IHttpClient.ts";
import {StatisticsPerDaysRequest} from "./models/statistics/StatisticsPerDaysRequest.ts";
import {StatisticsPerPeriodRequest} from "./models/statistics/StatisticsPerPeriodRequest.ts";

export class StatisticsClient {
  private readonly client: IHttpClient;

  public constructor(client: IHttpClient) {
    this.client = client;
  }

  public async perDays(request: StatisticsPerDaysRequest): Promise<Response> {
    return await this.client.get({url: "api/v1/statistics/per-days", query: request});
  }

  public async perPeriod(request: StatisticsPerPeriodRequest): Promise<Response> {
    return await this.client.get({url: "api/v1/statistics/per-period", query: request.period});
  }
}